.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,33 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        # Fetch all the history so the pre-commit/action can compare changes between the target and the git commit
+        fetch-depth: 0
+
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
+
+    - name: Run pre-commit on PRs
+      uses: pre-commit/action@v3.0.0
+      if: ${{ github.event.pull_request }}
+      with:
+        extra_args: --from-ref ${{ github.event.pull_request.base.sha }} --to-ref ${{ github.event.pull_request.head.sha }}
+
+    - name: Run pre-commit on branches
```

Comment:
We might need to compare with the previous commit too, otherwise, if there are linting errors they will be shown and report GitHub check failures